source/extensions/filters/http/admission_control/config.cc

```diff
@@ -0,0 +1,65 @@
+#include "extensions/filters/http/admission_control/config.h"
+
+#include "envoy/extensions/filters/http/admission_control/v3alpha/admission_control.pb.h"
+#include "envoy/extensions/filters/http/admission_control/v3alpha/admission_control.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "common/common/enum_to_int.h"
+
+#include "extensions/filters/http/admission_control/admission_control.h"
+#include "extensions/filters/http/admission_control/evaluators/default_evaluator.h"
+#include "extensions/filters/http/admission_control/evaluators/response_evaluator.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AdmissionControl {
+
+static constexpr std::chrono::seconds defaultSamplingWindow{120};
+
+Http::FilterFactoryCb AdmissionControlFilterFactory::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::admission_control::v3alpha::AdmissionControl& config,
+    const std::string& stats_prefix, Server::Configuration::FactoryContext& context) {
+
+  std::string prefix = stats_prefix + "admission_control.";
+
+  // Create the thread-local controller.
+  auto tls = context.threadLocal().allocateSlot();
+  auto sampling_window = std::chrono::seconds(
+      PROTOBUF_GET_MS_OR_DEFAULT(config, sampling_window, 1000 * defaultSamplingWindow.count()) /
+      1000);
+  tls->set(
+      [sampling_window, &context](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
+        return std::make_shared<ThreadLocalControllerImpl>(context.timeSource(), sampling_window);
+      });
+
+  std::unique_ptr<ResponseEvaluator> response_evaluator;
+  switch (config.evaluation_criteria_case()) {
+  case AdmissionControlProto::EvaluationCriteriaCase::kDefaultEvalCriteria:
+    response_evaluator = std::make_unique<DefaultResponseEvaluator>(config.default_eval_criteria());
+    break;
+  case AdmissionControlProto::EvaluationCriteriaCase::EVALUATION_CRITERIA_NOT_SET:
+  default:
```

Comment:
I don't think you need a default?

Reply:
I don't, but this catches a scenario where a new evaluator is implemented and configured, but not handled here. Otherwise, we'd catch it by dereferencing a null pointer somewhere.

Reply:
You will only allow the compiler to catch it if you remove default, which is what you want.